### DISCUSSION TITLE
Licensing

### DISCUSSION THREAD

**Discussion by bglw** [wailsapp/wails#755] [General]
Hi Lea! 🙂

Just a query on where the licensing will end up. Right now the project is labeled as MIT, but it looks like there are a range of incompatible licenses within, at least as far as <a href="https://app.fossa.io/projects/git%2Bgithub.com%2Fwailsapp%2Fwails?ref=badge_shield" alt="FOSSA Status"><img src="https://app.fossa.io/api/projects/git%2Bgithub.com%2Fwailsapp%2Fwails.svg?type=shield"/></a> is concerned. I don't know if that's an accurate summary of the distributed code, however.

What are the longer-term plans here?

**Comment by leaanthony** [comment]
Hi Liam,

Thanks for raising this. A while back, it was one FOSSA flag that was an
invalid match to an otf font. I raised this with FOSSA but they were
reluctant to change it. So I kinda mentally switched off to the widget. It
concerns me for sure and I'll look at removing or replacing license clashes.

That being said, for a tool like this, it feels like a more restrictive
license would benefit the project and include a clause that the templates
and runtime are excempt. I want to do the right thing by everyone else and
also myself. Am open to suggestions on how to achieve that.

Cheers,

-Lea.

**Reply by bglw** [reply]
If you're looking for a similar license to benefit the project, I'm a fan of the [MPL license](https://www.mozilla.org/en-US/MPL/2.0/FAQ/). It's fairly similar to Apache/MIT licenses in the permitted usage, but it requires that any changes to the codebase are open-sourced. Unlike the GPL family, it doesn't affect any other files outside of Wails within a larger codebase.

I would also totally understand going for a more restrictive or commercial license from your end.
